src/telemetry/mod.rs

```diff
@@ -0,0 +1,108 @@
+mod lang_exporter;
+pub(crate) mod metrics;
+
+use crate::telemetry::lang_exporter::{LangSpanExporter, OTelExportStreams};
+use itertools::Itertools;
+use opentelemetry::sdk::trace::Config;
+use opentelemetry::sdk::Resource;
+use opentelemetry::{
+    global, sdk::trace::TracerProvider, trace::TracerProvider as TracerProviderTrait, KeyValue,
+};
+use std::{collections::VecDeque, sync::Once};
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt, EnvFilter};
+
+const TELEM_SERVICE_NAME: &str = "temporal-core-sdk";
+const ENABLE_OPENTELEM_ENV_VAR: &str = "TEMPORAL_ENABLE_OPENTELEMETRY";
+static TRACING_INIT: Once = Once::new();
+
+/// Initialize tracing subscribers and output. Core will not call this itself, it exists here so
```

Comment:
Why don't we pass all necessary parameters into core init and initialize tracing from there?